### PR TITLE
make the HOST variable configurable

### LINKED ISSUE #5517: [Bug]: k8s dual stack IPv4/6 helm chart v0.1.413 and later readiness probe
### What happened?

When I was using Helm Chart v0.1.412, everything was good. But once I tried to update to any version after that, i.e., starting with 0.1.413 but also with 0.1.427, my pods no longer could start correctly. Note that the old helm chart v0.1.412 works fine with the newer version of the app main-v1.44.17-stable.
I have a DualStack k8s cluster with both IPv4 & IPv6 assigned to the LiteLLM pods. While in the logs inside the container, I see it assuming only IPv6.

I believe the issue is that the pod inside can see only IPv6 while the readiness probe is trying to reach IPv4.
Getting this error on the pod:
`Startup probe failed: Get "http://10.42.7.177:4000/health/readiness": dial tcp 10.42.7.177:4000: connect: connection refused`

**I believe the helm chart and probably the LiteLLM container need a fix to be able to run properly in the DualStack k8s clusters properly.**

I have created a kustomization to patch service manifests to include DualStack like this it works with 0.1.412 but it doesn't work for the newer versions with this patch or without this patch.
```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

- name: litellm-helm
  repo: oci://ghcr.io/berriai
  version: 0.1.413
  releaseName: litellm-proxy

patches:
  - op: replace
    path: /spec/ipFamilies
    value:
      - IPv4
      - IPv6
  - op: replace
    path: /spec/ipFamilyPolicy
    value:
      PreferDualStack
```

### Relevant log output

logs of the pod. Note the pod is running on `http://[::]:4000` meaning it assumed only the IPv6 address. 
While the readiness probe is configured for IPv4. This mismatch probably is the root cause of the problem.

```shell
✔ Generated Prisma Client Python (v0.11.0) to ./../../prisma in 301ms

INFO:     Started server process [1]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://[::]:4000 (Press CTRL+C to quit)
```
```


### Twitter / LinkedIn details

_No response_

### PR DESCRIPTION
**Pull request by menardorama**
## make the HOST variable configurable

Kubernetes clusters can be single stack or dual stack. This change address the recent change where dual stack cluster is set to be the default.

Setting can now be configurable in the chart

## Relevant issues
Fixes #5517

## Type
🐛 Bug Fix

## Changes

Add a value for the HOST env variable in order to make the helm chart less restrictive on the environment where it is deployed

## [REQUIRED] Testing - Attach a screenshot of any new tests passing locall
helm template . --set listen="0.0.0.0"
```
      containers:
        - name: litellm
          securityContext:
            {}
          image: "ghcr.io/berriai/litellm-database:main-v1.43.18"
          imagePullPolicy: IfNotPresent
          env:
            - name: HOST
              value: "0.0.0.0"
            - name: PORT
              value: "4000"
```

helm template .
```
      containers:
        - name: litellm
          securityContext:
            {}
          image: "ghcr.io/berriai/litellm-database:main-v1.43.18"
          imagePullPolicy: IfNotPresent
          env:
            - name: HOST
              value: ":"
            - name: PORT
              value: "4000"
```

### CODE REVIEW

File: deploy/charts/litellm-helm/templates/deployment.yaml [hunk 1]
```diff
@@ -102,7 +102,7 @@ spec:
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           env:
             - name: HOST
-              value: "::"
+              value: "{{ .Values.listen | default "::" }}"
```
Comment (ishaan-jaff) [inline deploy/charts/litellm-helm/templates/deployment.yaml:105]: would this work for ipv4 addresses ?

**Review by ishaan-jaff — CHANGES_REQUESTED** [review]
had a follow up question

File: deploy/charts/litellm-helm/templates/deployment.yaml (hunk 1 above)
Comment (menardorama) [reply deploy/charts/litellm-helm/templates/deployment.yaml:105]: you just have to set the value to 0.0.0.0 for ipv4

Comment (laurb9) [reply deploy/charts/litellm-helm/templates/deployment.yaml:105]: I think the default should be the original `0.0.0.0`, as this PR addresses a regression introduced when it was replaced with `::`

**Comment by qdrddr** [comment]
are you sure the `listen: ""` in the values.yaml file should be under ingress and not the image? @menardorama  & @ishaan-jaff 

https://github.com/BerriAI/litellm/pull/5529/commits/bcdeb630b5de93b241658fe85e095186e9f4625d#diff-2105d8e1ded03b7b26c7c794db81f24dd466f6f793fd07a37120863950c2429dR59-R80

**Comment by ghost** [comment]
@menardorama @qdrddr @laurb9 is this PR ready for merge? Seems like a few open comments. Ideally we can have this as part of tonight's release train

**Comment by menardorama** [comment]
> are you sure the `listen: ""` in the values.yaml file should be under ingress and not the image? @menardorama  & @ishaan-jaff 
> 
> 
> 
> https://github.com/BerriAI/litellm/pull/5529/commits/bcdeb630b5de93b241658fe85e095186e9f4625d#diff-2105d8e1ded03b7b26c7c794db81f24dd466f6f793fd07a37120863950c2429dR59-R80

I don't understand ....

You mean the place in the values ? Listen address has nothing to do with ingress but whatever as soon as it fix the issue.....
Actually my ci is patching the deployment after helm install I. Order to address the issue.

**Comment by qdrddr** [comment]
> > are you sure the `listen: ""` in the values.yaml file should be under ingress and not the image? @menardorama  & @ishaan-jaff
> > [bcdeb63#diff-2105d8e1ded03b7b26c7c794db81f24dd466f6f793fd07a37120863950c2429dR59-R80](https://github.com/BerriAI/litellm/commit/bcdeb630b5de93b241658fe85e095186e9f4625d#diff-2105d8e1ded03b7b26c7c794db81f24dd466f6f793fd07a37120863950c2429dR59-R80)
> 
> I don't understand ....
> 
> You mean the place in the values ? Listen address has nothing to do with ingress but whatever as soon as it fix the issue..... Actually my ci is patching the deployment after helm install I. Order to address the issue.

Yes. I was asking if you have selected the correct placement for the variable in the values file. Currently the `listen` is placed under ingress. If you click on the link you'll see it is placed under the ingress. Shouldn't it be under the image item?

https://github.com/BerriAI/litellm/commit/bcdeb630b5de93b241658fe85e095186e9f4625d#diff-2105d8e1ded03b7b26c7c794db81f24dd466f6f793fd07a37120863950c2429dR59-R80

CleanShot 2024-09-06 at 09 08 19

**Comment by laurb9** [comment]
> Yes. I was asking if you have selected the correct placement for the variable in the values file. Currently the `listen` is placed under ingress. If you click on the link you'll see it is placed under the ingress. Shouldn't it be under the image item?

It is actually at root level like masterkey, not under ingress. Github misinterprets where the comment is located based on the last key it saw.

**Comment by menardorama** [comment]
> > Yes. I was asking if you have selected the correct placement for the variable in the values file. Currently the `listen` is placed under ingress. If you click on the link you'll see it is placed under the ingress. Shouldn't it be under the image item?
> 
> It is actually at root level like masterkey, not under ingress. Github misinterprets where the comment is located based on the last key it saw.

Exactly, now maybe it can be a good idea to have a dedicated section where masterkey, listen address and future litellm runtime settings are located together ?

**Comment by laurb9** [comment]
> Exactly, now maybe it can be a good idea to have a dedicated section where masterkey, listen address and future litellm runtime settings are located together ?

Something like proxy config ? moving masterkey would be a breaking change and out of scope though.

**Comment by menardorama** [comment]
Exactly....

Even if the correct way would be that....

What about a section called runtimeConfig ?

If things won't move a lot on the runtime part this can be let on the root level

**Comment by ghost** [comment]
Hey @menardorama @laurb9 @qdrddr is this PR ready for review / merge? 

If there's additional improvements we can make, would it help to scope that into a separate PR?

**Comment by menardorama** [comment]
In my opinion having to change where the master key setting is defined will have a huge impact on all deployments.

So for now adding another setting at the root level may be the best solution for now.

Now regarding the default value, is having 0.0.0.0 as default and permit to define it to "::" may be the best choice.

If yes, I can update my PR and you can merge it

Is this plan is ok for you ?

**Comment by ghost** [comment]
Hey @menardorama i merged in #5587 which should address the host name issue. Closing this PR as it seemed to address the same issue. If you'd like to submit another PR for improvements on the existing helm chart, that would be great

**Comment by menardorama** [comment]
> Hey @menardorama i merged in #5587 which should address the host name issue. Closing this PR as it seemed to address the same issue. If you'd like to submit another PR for improvements on the existing helm chart, that would be great

Hi, 
No problem but new settings needs to be defined in values.yaml even with only comments; this is to avoid having to deep dive  in the helm chart to understand and find this setting....

**Pull request closed without merging**